src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,45 @@
 
 class Animal {
   // write your code here
+  // static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  // alive.push(this);
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+
+  bite(herbivore) {
+    if (herbivore.hidden === false) {
+      herbivore.health -= 50;
+    }
+
+    if (herbivore.health <= 0) {
+      const index = Animal.alive.indexOf(herbivore);
+
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
Instead of using .indexOf() and .splice(), you can just use the .filter() method

Reply:
Depends...